examples/parse_args.h

```diff
@@ -0,0 +1,213 @@
+    for (int i = 1; i < argc; i++) {
+        if (argv[i] && argv[i][0] == '-') {
+            return argv[i];
+        }
+    }
+    return NULL;
+}
+
+/**
+ * Parse positional arguments from `argv`. Must be called after all expected options are parsed, and after checking that
+ * no unknown options remain in `argv`
+ * @param argc
+ * @param argv
+ * @param nb_args: number of expected positional arguments
+ * @returns NULL if found more positional arguments than `nb_args`. Else an array of found arguments in order, followed
+ * by NULL values if found less positional arguments than `nb_args`
+ * @note Returned pointer is dynamically allocated and must be freed
+ */
+char** parse_pos_args(const int argc, char** argv, const size_t nb_args) {
+    char** pos_argv = (char**)calloc(nb_args, sizeof(char*));
+    size_t pos_argc = 0;
+    for (int i = 1; i < argc; i++) {
+        if (argv[i]) {
+            pos_argc++;
+            if (pos_argc > nb_args) {
+                free(pos_argv);
+                return NULL;
+            }
+            pos_argv[pos_argc - 1] = argv[i];
+        }
+    }
+    return pos_argv;
+}
+
+/**
+ * Parse zenoh options that require a JSON-serialized list (-e, -l from common args) and add them to
+ * `config`. Prints error message and exits if fails to insert parsed values
+ * @param argc
+ * @param argv
+ * @param opt: option to parse (without `-` or `--` prefix)
+ * @param config: address of an owned zenoh configuration
+ * @param config_key: zenoh configuration key under which the parsed values will be inserted
+ */
+void parse_zenoh_json_list_config(int argc, char** argv, const char* opt, const char* config_key,
+                                  const z_owned_config_t* config) {
+    char* buf = (char*)calloc(1, sizeof(char));
+    const char* value = parse_opt(argc, argv, opt, true);
+    while (value) {
+        size_t len_newbuf = strlen(buf) + strlen(value) + 4;  // value + quotes + comma + nullbyte
+        char* newbuf = (char*)malloc(len_newbuf);
+        snprintf(newbuf, len_newbuf, "%s'%s',", buf, value);
+        free(buf);
+        buf = newbuf;
+        value = parse_opt(argc, argv, opt, true);
+    }
+    size_t buflen = strlen(buf);
+    if (buflen > 0) {
+        // remove trailing comma
+        buf[buflen - 1] = '\0';
+        buflen--;
+        // add list delimiters
+        size_t json_list_len = buflen + 3;  // buf + brackets + nullbyte
+        char* json_list = (char*)malloc(json_list_len);
+        snprintf(json_list, json_list_len, "[%s]", buf);
+        // insert in config
+        if (zc_config_insert_json(z_loan(*config), config_key, json_list) < 0) {
+            printf(
+                "Couldn't insert value `%s` in configuration at `%s`\n`%s` is either not a JSON-serialized list of "
+                "strings, or values within the list do not respect expected format for `%s`\n",
+                json_list, json_list, json_list, config_key);
```

Comment:
```suggestion
                json_list, config_key, json_list, config_key);
```

Reply:
The rest of the message applies to the value given by the user, not the key under which it is inserted. I feel like the way it is right now is clearer, or else it must be adapted to say "must be a JSON-serialized list of strings, and values within the list..."